### PR TITLE
Reduce number of false postive scraper results

### PR DESCRIPTION
**Pull request by simplay**
Partially addresses Issue #16 

The general idea is to only show results that exhibit a partial substring match.
Moreover, the matches are now case-insensitive and delimiter tokens (e.g. `-`, ` `, `/`) have no influence on the match.

For example, with this patch, the following 3 queries become the same: `He Man`, `he man` and `He-Man`

**before**:

before

**after**:

after

## TODO

+ [x] Better naming + Docs
+ [x] Decrypt Logic
+ [x] Write tests

### CODE REVIEW

**Comment by simplay** [comment]
@rathrio: This PR is ready for a review. 👍

File: app/scrapers/fuzzy_substring_match_filter.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    tmp
+  end
+
+  # Obtain all substring combinations of a given word.
+  #
+  # @example
+  #   substrings_for("abc")
+  #   #=> ["a", "ab", "abc", "b", "bc", "c"]
+  #
+  # @param s [String] word to split
```
Comment (rathrio) [inline app/scrapers/fuzzy_substring_match_filter.rb]: s -> word?

File: app/scrapers/fuzzy_substring_match_filter.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Forward only results that exhibit a substring match with
+# a given query word.
+#
+# This filter is used to reduce the number of false positive results provided
+# by our scappers and is used within the ScraperResultsController class.
```
Comment (rathrio) [inline app/scrapers/fuzzy_substring_match_filter.rb]: scappers -> scrapers

File: test/scrapers/fuzzy_substring_match_filter_test.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  test 'Only matches partial substrings' do
+    results = [
+      { name: "He Man: Vol 1" },
+      { name: "The He Man: Vol 1" },
+      { name: "Her Men: Vol 1" },
+      { name: "Simpons - He is the real man" }
+    ]
+
+    gt = [
```
Comment (rathrio) [inline test/scrapers/fuzzy_substring_match_filter_test.rb]: I'm having trouble figuring out what `gt` stands for 😅 ...Can you write it out?

File: test/scrapers/fuzzy_substring_match_filter_test.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    substrings.each do |split|
+      assert gt_splits.include?(split)
+    end
+
+    gt_splits.each do |split|
+      assert substrings.include?(split)
+    end
+  end
+
+end
```
Comment (rathrio) [inline test/scrapers/fuzzy_substring_match_filter_test.rb]: Really comprehensive tests! 👍 👍

File: app/scrapers/fuzzy_substring_match_filter.rb [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+class FuzzySubstringMatchFilter
```
Comment (rathrio) [inline app/scrapers/fuzzy_substring_match_filter.rb]: As discussed, move to `app/filters`.

**Review by rathrio — CHANGES_REQUESTED** [review]
This is really cool. Thank you!

**Comment by simplay** [comment]
PR is ready to go.

**Review by rathrio — APPROVED** [review]

**Comment by rathrio** [comment]
👍

**Pull request merged**